### PR TITLE
Added support for middleware topology page

### PR DESCRIPTION
**Pull request by jkandasa**
Added new `Mixin` --> `TopologyMixin`. With this `Mixin` we can perform following actions,

* Get legends
* Get elements with filter support(optional)
* Get parents and children of an element
* Do actions on `Search box`, `Display Names`, `Legends`
* Perform `double_click` on an element

{{pytest: cfme/tests/middleware/ -v --long-running --use-provider hawkular}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 2de5fc58d8b63ecd47a8bd0c61543a53d8c3cdb6:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by jkandasa** [comment]
@psav @mfalesni @jkrocil Can you please review this PR?

**Comment by psav** [comment]
Sure

**Comment by psav** [comment]
Initial question, could/should this most of this live in cfme/web_ui in a new module, We have had things like the timelines living in here, as this is where UI widget interaction lives?

**Comment by jkandasa** [comment]
@psav I felt like `cfme/commons` will be a good place for `Topology` page. We can change the location if needed.

**Comment by jkrocil** [comment]
I agree with @psav here. The `Topology` and related classes belong under `cfme.web_ui.topology`.

**Comment by jkandasa** [comment]
@psav @jkrocil Thank you! I have changed the code as per your comments.

**Comment by dajoRH** [comment]
Lint report for commit bccbf432cd6f41c76eadb97fed872cb12049815a:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/middleware/provider.py [hunk 1]
```diff
@@ -174,3 +175,7 @@ def load_details(self, refresh=False):
             tmp_provider = _db_select_query(
                 name=self.name, type='ManageIQ::Providers::Hawkular::MiddlewareManager').first()
             self.db_id = tmp_provider.id
+
+    def load_topology_page(self):
+        self.load_details()
```
Comment (mfalesni) [inline cfme/middleware/provider.py]: I would prefer self.summary.reload() in this case.

Using summary implies load_details. But it is more importnat to make sure the summary is fresh and we are on the page which this ensures.

Comment (jkandasa) [inline cfme/middleware/provider.py]: @mfalesni Thank you! I have changed to `self.summary.reload()`

**Comment by dajoRH** [comment]
Lint report for commit 14c0cc6c5d0b8cab5bbcb83aef343cbb0aed558a:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 01c102fa7bc68fb90270cf586842663d35585910:

Everything seems all right :smile: .
*CFME QE Bot*

**Pull request merged**